src/borg/helpers.py

```diff
@@ -773,6 +773,14 @@ def parse(self, text):
         if not valid:
             return False
         self.archive = m.group('archive')
+        return self._is_archive_name_printable()
+
+    # lookout for non-printable characters
+    def _is_archive_name_printable(self):
+        if self.archive:
+            for char in self.archive:
+                if unicodedata.category(char)[0] == 'C':
+                    return False
         return True
```

Comment:
this shouldn't be a method, but rather a global utility function.

implementation could looks somehow like:

def printable(s):
    return all(unicodedata.category(c)[0] == 'C' for c in s)

wondering if such a function does not already exist? in stdlib?